cpp/oneapi/dal/algo/kmeans/backend/gpu/train_kernel_lloyd_dense_dpc.cpp

```diff
@@ -52,17 +52,15 @@ static NumericTablePtr get_initial_centroids(const dal::backend::context_gpu& ct
 
     const auto data = input.get_data();
 
-    const int64_t row_count = data.get_row_count();
     const int64_t column_count = data.get_column_count();
-
     const int64_t cluster_count = params.get_cluster_count();
 
     daal::data_management::NumericTablePtr daal_initial_centroids;
 
     if (!input.get_initial_centroids().has_data()) {
-        auto arr_data = row_accessor<const Float>{ data }.pull(queue);
-        const auto daal_data =
-            interop::convert_to_daal_sycl_homogen_table(queue, arr_data, row_count, column_count);
+        // We use CPU algorithm for initialization, so input data
+        // may be copied to DAAL homogen table
+        const auto daal_data = interop::copy_to_daal_homogen_table<Float>(data);
```

Comment:
`copy_to_daal_homogen_table` uses `row_accessor::pull`, it copies data to host if necessary.

The other case is `convert_to_daal_table`, it uses adapter, so it may require copy to host. It affects the case when we run device algorithm and then pass its result to host algorithm. There is a bunch of problems here:
- Host variant of `convert_to_daal_table` does not support device -> host conversion if it goes through adapter branch.
- There is no integration testing for that case, we need to have at least two small test cases:
   - Pass device data to host algorithm
   - Pass pure host data to device algorithm   

I can fix current version of `convert_to_daal_table` as part of this PR, but we need to have more solid solution. This requires:
- Refactoring of table/accessor implementation to simplify. The solution requires adding new methods to table iface, so it may be problematic.
- Add `to_host`/`to_device` for table, as we have for array.